### ISSUE TITLE
Implement mixin application inference behind the super-mixins flag

### ISSUE THREAD

**Issue by a-siva** [dart-lang/sdk#32019]
In order to land the flutter framework changes suggested in #31984 (issues with mixins) the analyzer needs to support mixin application inference behind the super-mixins flag.

The pull request for flutter https://github.com/flutter/flutter/pull/14392 is blocked on this.

**Comment by a-siva** [comment]
Analyzer produces the following errors with the PR referenced above

```
[lint] Specify type annotations. (dev/manual_tests/lib/material_arc.dart, line 419, col 61)
[lint] Specify type annotations. (dev/manual_tests/lib/animated_icons.dart, line 39, col 60)
[lint] Specify type annotations. (dev/manual_tests/lib/text.dart, line 119, col 47)
[lint] Specify type annotations. (dev/manual_tests/lib/text.dart, line 793, col 45)
[lint] Specify type annotations. (dev/manual_tests/lib/text.dart, line 900, col 51)
[lint] Specify type annotations. (dev/integration_tests/external_ui/lib/main.dart, line 26, col 44)
[lint] Specify type annotations. (examples/layers/services/isolate.dart, line 207, col 62)
[lint] Specify type annotations. (examples/layers/widgets/spinning_square.dart, line 12, col 63)
[lint] Specify type annotations. (examples/flutter_gallery/lib/demo/material/tabs_fab_demo.dart, line 42, col 57)
[lint] Specify type annotations. (examples/flutter_gallery/lib/demo/material/progress_indicator_demo.dart, line 14, col 77)
[lint] Specify type annotations. (examples/flutter_gallery/lib/demo/material/scrollable_tabs_demo.dart, line 43, col 70)
[lint] Specify type annotations. (examples/flutter_gallery/lib/demo/material/grid_list_demo.dart, line 63, col 65)
[lint] Specify type annotations. (examples/flutter_gallery/lib/demo/material/bottom_navigation_demo.dart, line 87, col 10)
[lint] Specify type annotations. (examples/flutter_gallery/lib/demo/material/drawer_demo.dart, line 19, col 55)
[lint] Specify type annotations. (examples/flutter_gallery/lib/demo/video_demo.dart, line 228, col 10)
[lint] Specify type annotations. (examples/flutter_gallery/lib/demo/video_demo.dart, line 364, col 10)
[lint] Specify type annotations. (examples/flutter_gallery/lib/gallery/home.dart, line 111, col 56)
[lint] Specify type annotations. (examples/catalog/lib/app_bar_bottom.dart, line 12, col 71)
[lint] Specify type annotations. (packages/flutter/test/material/tabs_test.dart, line 87, col 70)
[lint] Specify type annotations. (packages/flutter/test/material/tabbed_scrollview_warp_test.dart, line 31, col 55)
[lint] Specify type annotations. (packages/flutter/test/widgets/ticker_provider_test.dart, line 72, col 67)
[lint] Specify type annotations. (packages/flutter/test/widgets/animated_cross_fade_test.dart, line 373, col 76)
[lint] Specify type annotations. (packages/flutter/lib/src/material/two_level_list.dart, line 150, col 65)
[lint] Specify type annotations. (packages/flutter/lib/src/material/app_bar.dart, line 922, col 59)
[lint] Specify type annotations. (packages/flutter/lib/src/material/drawer.dart, line 161, col 66)
[lint] Specify type annotations. (packages/flutter/lib/src/material/ink_well.dart, line 377, col 70)
[lint] Specify type annotations. (packages/flutter/lib/src/material/switch.dart, line 126, col 47)
[lint] Specify type annotations. (packages/flutter/lib/src/material/input_decorator.dart, line 1331, col 63)
[lint] Specify type annotations. (packages/flutter/lib/src/material/input_decorator.dart, line 125, col 66)
[lint] Specify type annotations. (packages/flutter/lib/src/material/input_decorator.dart, line 239, col 58)
[lint] Specify type annotations. (packages/flutter/lib/src/material/popup_menu.dart, line 352, col 95)
[lint] Specify type annotations. (packages/flutter/lib/src/material/stepper.dart, line 182, col 49)
[lint] Specify type annotations. (packages/flutter/lib/src/material/radio.dart, line 105, col 51)
[lint] Specify type annotations. (packages/flutter/lib/src/material/scrollbar.dart, line 55, col 53)
[lint] Specify type annotations. (packages/flutter/lib/src/material/text_field.dart, line 282, col 53)
[lint] Specify type annotations. (packages/flutter/lib/src/material/tab_controller.dart, line 288, col 75)
[lint] Specify type annotations. (packages/flutter/lib/src/material/bottom_navigation_bar.dart, line 274, col 73)
[lint] Specify type annotations. (packages/flutter/lib/src/material/checkbox.dart, line 99, col 51)
[lint] Specify type annotations. (packages/flutter/lib/src/material/expand_icon.dart, line 61, col 55)
[lint] Specify type annotations. (packages/flutter/lib/src/material/mergeable_material.dart, line 153, col 69)
[lint] Specify type annotations. (packages/flutter/lib/src/material/refresh_indicator.dart, line 136, col 66)
[lint] Specify type annotations. (packages/flutter/lib/src/material/expansion_tile.dart, line 82, col 61)
[lint] Specify type annotations. (packages/flutter/lib/src/material/material.dart, line 252, col 51)
[lint] Specify type annotations. (packages/flutter/lib/src/material/time_picker.dart, line 1001, col 44)
[lint] Specify type annotations. (packages/flutter/lib/src/material/slider.dart, line 172, col 47)
[lint] Specify type annotations. (packages/flutter/lib/src/material/tooltip.dart, line 97, col 49)
[lint] Specify type annotations. (packages/flutter/lib/src/material/data_table.dart, line 682, col 54)
[lint] Specify type annotations. (packages/flutter/lib/src/material/progress_indicator.dart, line 352, col 85)
[lint] Specify type annotations. (packages/flutter/lib/src/material/progress_indicator.dart, line 172, col 81)
[lint] Specify type annotations. (packages/flutter/lib/src/material/scaffold.dart, line 544, col 50)
[lint] Specify type annotations. (packages/flutter/lib/src/material/scaffold.dart, line 182, col 96)
[lint] Specify type annotations. (packages/flutter/lib/src/rendering/proxy_box.dart, line 37, col 84)
[lint] Specify type annotations. (packages/flutter/lib/src/widgets/scrollable.dart, line 252, col 54)
[lint] Specify type annotations. (packages/flutter/lib/src/widgets/animated_list.dart, line 209, col 58)
[lint] Specify type annotations. (packages/flutter/lib/src/widgets/overlay.dart, line 522, col 49)
[lint] Specify type annotations. (packages/flutter/lib/src/widgets/overlay.dart, line 266, col 48)
[lint] Specify type annotations. (packages/flutter/lib/src/widgets/animated_cross_fade.dart, line 232, col 69)
[lint] Specify type annotations. (packages/flutter/lib/src/widgets/implicit_animations.dart, line 236, col 98)
[lint] Specify type annotations. (packages/flutter/lib/src/widgets/navigator.dart, line 740, col 52)
[lint] Specify type annotations. (packages/flutter/lib/src/widgets/overscroll_indicator.dart, line 129, col 87)
[lint] Specify type annotations. (packages/flutter/lib/src/widgets/dismissible.dart, line 192, col 83)
[lint] Specify type annotations. (packages/flutter/lib/src/widgets/dismissible.dart, line 192, col 57)
[lint] Specify type annotations. (packages/flutter/lib/src/widgets/fade_in_image.dart, line 351, col 57)
[lint] Specify type annotations. (packages/flutter/lib/src/widgets/editable_text.dart, line 311, col 58)
[lint] Specify type annotations. (packages/flutter/lib/src/cupertino/switch.dart, line 98, col 65)
[lint] Specify type annotations. (packages/flutter/lib/src/cupertino/button.dart, line 114, col 65)
[lint] Specify type annotations. (packages/flutter/lib/src/cupertino/scrollbar.dart, line 54, col 71)
[lint] Specify type annotations. (packages/flutter/lib/src/cupertino/activity_indicator.dart, line 36, col 87)
[lint] Specify type annotations. (packages/flutter/lib/src/cupertino/slider.dart, line 122, col 65)
[lint] 2 public members lack documentation (ran in 48.8s)
```

**Comment by devoncarew** [comment]
@a-siva, I'm going to assign this to @stereotype441. It looks like an active discussion in https://github.com/dart-lang/sdk/issues/31984; is the correct solution for this agreed on?

If the flutter PR is blocked on this, you could look into disabling the lint for `Specify type annotations.`. That would be done here I believe https://github.com/flutter/flutter/blob/master/analysis_options_repo.yaml#L47.

**Comment by a-siva** [comment]
I believe Lasse and Leaf have an agreement, see the email thread from Lasse this morning.
I do not feel comfortable disabling a lint just to land this because it will eventually lead to a runtime error anyways.

**Comment by stereotype441** [comment]
I've just had about an hour long IM with @leafpetersen and it sounds like there is *not* an agreement yet.  Leaf is going to follow up (either here or on the internal email thread) with details.  I realize this is a P0 bug but I can't take action without a clear direction from the language team, so for the moment I'm going to reassign this bug to them.

**Comment by leafpetersen** [comment]
> If the flutter PR is blocked on this, you could look into disabling the lint for Specify type annotations.. That would be done here I believe

I believe @Hixie requested that these classes be marked with @optionalTypeArgs for this reason.

**Comment by leafpetersen** [comment]
Is the analyzer implementation of this being worked on?  @devoncarew @bwilkerson

**Comment by leafpetersen** [comment]
Looks like this is at least partially implemented?  Should this be closed, or is it still open pending further iteration?

**Comment by stereotype441** [comment]
Regarding the analyzer implementation, I am working on it.  I've filed a separate issue to track the analyzer work: #32146

**Comment by vsmenon** [comment]
@kmillikin implemented this in CFE.  @stereotype441 is tracking the analyzer impl in #32146, so closing this one.

**Comment by kmillikin** [comment]
One thing to note: this is not yet behind a flag, it is on for Fasta with --strong and can't be disabled.  I'll change that next week by exposing an option in the Fasta API for the VM and plumbing through the flag from the VM.
